DEV-CONVENTIONS.md

```diff
@@ -36,6 +36,8 @@ Others docs you should look at:
 
       - Must use lowerCamel case for Go package names.
 
+      - Must not use camel case for file names, e.g. type `DeviceAllocator` should reside in `device_allocator.go` or in `deviceallocator.go`.
```

Comment:
please qualify. This should be applicable only to go files. We do have no go files in this repo as well (samples). They should follow their language conventions.

Reply:
@rshriram Done.